### DISCUSSION TITLE
Support LUNs for CD-ROM or RO type activity.

### DISCUSSION THREAD

**Discussion by TheBajaGuy** [BlueSCSI/BlueSCSI-v2#145] [Ideas]
Enhancement Idea

In an online form discussion with a user of BlueSCSI v2 last month, they were trying to figure out how to change CD media images on the unit.  I didn't have any experience with the BlueSCSI v2 unit (I had just bought one, but had yet to set it up).  I have used Zulu and SCSI2SD v5/v6 units, a MonsterCF, and a SCSI2SATA in recent times, so I am familiar with more than some of the details.

During that discussion, it occurred to me that many forget, or may be unaware, that the SCSI spec supports LUNs (Logical Unit Numbers) off of each SCSI ID.  For 8-bit SCSI, that's sub-devices 0-7, with the device root typically at LUN 0 or LUN 7.  For devices without any LUNs, the LUN value is 0.  Multiple media/loader CD/DVD changer units typically have done things with LUNs, like Pioneer, using one of the LUNs as the control address for the media select/servo controls and lock/eject.  For R/W removable devices, the Bernoulli removable media drives I believe used LUNs 0 and 1 for the media on each side (it's been 35 years, so that might not b 100% accurate), along with some ST-506, IDE, and other disk interface to SCSI adapter boards.  

I have since run across the BlueSCSI ToolBox Amiga project here on Github.  I provides a means to select CD media on the unit with a tool from the Amiga host.  It looks interesting, and I will look at it in the near future.  At the same time, it might be more robust to support something like a LUN 0 or LUN 7 as the control interface, and the remaining 7 LUNs as slots for the CD r/o media.  The LUN value for the controller interface could be given a control file which is then maintained to defined the active images at the various LUNs.  It could enumerate all of the CD-type media.

Hope this at least spurns some further thought.

**Comment by zefrenchtoon** [comment]
There is another thread on this subject here:

https://github.com/BlueSCSI/BlueSCSI-v2/discussions/139
